utilities/shared_disk.py

```
@@ -253,58 +296,644 @@ def verify_shared_disk_data(
+        owner_vm: PyVmomi VM object for the owner VM.
+        owner_name: Display name of the owner VM.
+        source_provider_data: Provider config (uses guest_vm_win_user/password).
+
+    Returns:
+        Tuple of (NamePasswordAuthentication, vcenter_host).
+
+    Raises:
+        ValueError: If Tools not ready or vCenter host unavailable.
+    """
+    if not source_provider.wait_for_vmware_guest_info(owner_vm, timeout=_GUEST_TOOLS_READY_TIMEOUT):
+        raise ValueError(f"VMware Tools not available on '{owner_name}' after power-on, cannot label shared disk")
```

Ruff (0.15.18)

[warning] Avoid specifying long messages outside the exception class

(TRY003)